nanoPU_sim.py

```diff
@@ -435,33 +420,28 @@ def creditEvent(self, tx_msg_id, new_credit = None, dif_credit = None,
             elif opCode == None:
                 self.log('ERROR: creditEvent was triggered without a opCode value!')
             elif opCode == 'write':
-                if new_credit != None:
-                    if relOp(compVal, cur_credit):
-                        self.log('Changing credit for msg {} from {} to {}'.format(tx_msg_id, cur_credit, new_credit))
-                        self.credit[tx_msg_id] = new_credit
-                else:
-                    self.log("ERROR: creditEvent was triggered with 'write' opCode but no 'new_credit' value is provided!")
+                if relOp(compVal, cur_credit):
+                    self.log('Changing credit for msg {} from {} to {}'.format(tx_msg_id, cur_credit, new_credit))
+                    self.credit[tx_msg_id] = new_credit
             elif opCode == 'add':
-                if dif_credit != None:
-                    if relOp(compVal, cur_credit):
-                        self.credit[tx_msg_id] += dif_credit
-                        self.log('Changing credit for msg {} from {} to {}'.format(tx_msg_id, cur_credit, self.credit[tx_msg_id]))
-                else:
-                    self.log("ERROR: creditEvent was triggered with 'add' opCode but no 'dif_credit' value is provided!")
+                if relOp(compVal, cur_credit):
+                    self.credit[tx_msg_id] += new_credit
+                    self.log('Changing credit for msg {} from {} to {}'.format(tx_msg_id, cur_credit, self.credit[tx_msg_id]))
             elif opCode == 'shift_right':
-                if dif_credit != None:
-                    if relOp(compVal, cur_credit):
-                        self.credit[tx_msg_id] >>= dif_credit
-                        self.log('Changing credit for msg {} from {} to {}'.format(tx_msg_id, cur_credit, self.credit[tx_msg_id]))
-                else:
-                    self.log("ERROR: creditEvent was triggered with 'shift_right' opCode but no 'dif_credit' value is provided!")
+                if relOp(compVal, cur_credit):
+                    self.credit[tx_msg_id] >>= new_credit
+                    self.log('Changing credit for msg {} from {} to {}'.format(tx_msg_id, cur_credit, self.credit[tx_msg_id]))
             else:
                 self.log('ERROR: creditEvent was triggered for unknown opCode: {}'.format(opCode))
 
-            if self.toBtx[tx_msg_id] & (1<<self.credit[tx_msg_id])-1 != 0:
-                self.enq_active_messages_fifo(tx_msg_id)
-        else:
-            self.log('ERROR: creditEvent was triggered for unknown tx_msg_id: {}'.format(tx_msg_id))
```

Comment:
Are we getting rid of the ERROR prompt here? (It is okay to delete it in terms of functionality but I think it would be nice to see detailed logs during experiments.)

Reply:
The `creditToBtxEvent` is now invoked by both NACKs and PULLs. When a NACK invokes this action it will set `new_credit = None`.

Reply:
Right. Thanks. This else clause was for unknown tx_msg_id values. Maybe we could ad it as a separate if statement to check that. Leaving this for now. :)